docs/reference/mapping/types/version.asciidoc

```
@@ -0,0 +1,125 @@
+will all appear _after_ any valid version with regular alphabetical ordering. 
+
+[discrete]
+[[version-params]]
+==== Parameters for wildcard fields
+
+The following parameters are accepted by `version` fields:
+
+[horizontal]
+
+<<mapping-field-meta,`meta`>>::
+
+    Metadata about the field.
+
+[discrete]
+==== Limitations
+
+This field type isn't optimized for heavy wildcard, regex or fuzzy searches. While those
+type of queries work in this field, you should consider using a regular `keyword` field if
+you strongly rely on these kind of queries.
+
+==== Script support
+
+TBD: The `version` fields offers some specialized access to detailed information derived from
+valid version strings like the Major, Minor or Patch release number, whether the version value
+is valid according to Semver or if it is a pre-release version. This can be helpful when e.g.
+filtering for only released versions or running aggregations on parts of the version.
+The following query, for example, filters for released versions and groups them by Major version
+using a `terms` aggregation:
+
+[source,console]
+--------------------------------------------------
+POST my-index-000001/_search
+{
+  "query": {
+    "bool": {
+      "filter": [
+        {
+          "script": {
+            "script": {
+              "source": "doc['my_version'].isPreRelease() == false"
+            }
+          }
+        }
+      ]
+    }
+  },
+  "aggs": {
+    "group_major": {
+      "terms": {
+        "script": { "source": "doc['my_version'].getMajor()"},
+        "order": {
+          "_key": "asc"
+        }
+      }
+    }
+  }
+}
+
+--------------------------------------------------
+// TEST[continued]
+
+Functions available on via doc values in scripting are:
+
+[horizontal]
+
+isValid()::
+    Returns `true` if the field contains a version thats legal according to the Semantic Versioning rules
+
+isPreRelease()::
+    Returns `true` if the field contains a pre-release version
```

should we also add that `false` is returned if a version is invalid? Or may be these functions should fail if the version is invalid than it will be up to a user to guard agains this.

---

I'm thinking about switching the semantics here to `isRelease()` because that is more likely what users will filter for (e.g. show me all errors in released versions between 1.2.3 and 2.3.4). `isRelease()` then could be `false` for invalid versions and if you are interesed in the distinction you can always use `isValid()` with it. Would that make more sense to you?

---

I changed this in f1bf52f, let me know if you think this is better or if I should revert.

---

Thanks, having `isRelease()` function with semantics you defined looks very good to me.